R/simulationResultsToDataFrame.R

```diff
@@ -0,0 +1,136 @@
+#' Converts a `SimulationResults` objects to a data.frame
+#'
+#' @inheritParams getOutputValues
+#'
+#' @examples
+#' library(ospsuite)
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#'
+#' # Running an individual simulation
+#' # results is an instance of `SimulationResults`
+#' results <- runSimulations(sim)
+#'
+#' # convert to a dataframe
+#' simulationResultsToDataFrame(results)
+#' @return
+#'
+#' SimulationResults object as data.frame with columns IndividualId, Time, paths,
+#' simulationValues, unit, dimension, TimeUnit.
+#'
+#' @export
+
+simulationResultsToDataFrame <- function(simulationResults,
+                                         quantitiesOrPaths = NULL,
+                                         population = NULL,
+                                         individualIds = NULL) {
+  # no need to validating the input because this will be done by
+  # getOutputValues() anyways
+  simList <- getOutputValues(
+    simulationResults = simulationResults,
+    quantitiesOrPaths = quantitiesOrPaths,
+    population = population,
+    individualIds = individualIds
+  )
+
+  # convert data to long format with a new column for paths
+  df_data <- .convert_to_long(
+    simList$data,
+    cols = setdiff(names(simList$data), c("IndividualId", "Time")),
+    names_to = "paths",
+    values_to = "simulationValues"
+  )
+
+  # extract units and dimensions for paths in a separate dataframe
+  # iterate over the list (in .x) using index (names for list elements)
+  # and apply function in .f to each elements
+  # the result will be a list of dataframes, which will be bound into
+  # a single dataframe with the _dfr variant of this function
+  df_meta <- purrr::imap_dfr(
+    .x  = simList$metaData,
+    .f  = ~ as.data.frame(.x, row.names = NULL),
+    .id = "paths"
+  )
+
+  # leave out time units and dimensions since it is not a path
+  df_meta <- dplyr::filter(df_meta, paths != "Time")
+
+  # combine dataframe with simulated data and meta data
+  df <- dplyr::left_join(df_data, df_meta, by = "paths")
+
+  # add time unit to the this combined dataframe
+  # the time dimension is "Time", which is redundant and is thus left out
+  df <- dplyr::bind_cols(
+    df,
+    data.frame("TimeUnit" = simList$metaData$Time$unit[[1]])
+  )
+
+  # return the combined dataframe
+  return(df)
+}
+
+# custom function to convert dataframe from wide to long format
+# simplified version of
+# https://github.com/easystats/datawizard/blob/master/R/data_reshape.R
+# IP is author on this package
+
+.convert_to_long <- function(data,
```

Comment:
Why are we not relying on the packages imported (e..g the code you had yesterday?) How is this better?

Reply:
It's better in the sense that it only relies on base-R, while tidyr has a bunch of dependencies.

I am not getting rid of dplyr or purrr dependencies because I am also using them in the DataCombined branch.

Reply:
oh I see. So you removed one dependencies. Hard to tell from the flow. Thanks